ingest/boltcache_test.go

```diff
@@ -0,0 +1,109 @@
+package ingest
+
+import (
+	"encoding/gob"
+	"io/ioutil"
+	"net"
+	"os"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/gravwell/gravwell/v3/chancacher"
+	"github.com/gravwell/gravwell/v3/ingest/entry"
+)
+
+const (
+	tstLoc       string = `/tmp/cache_test.db`
+	memCacheSize uint64 = 1024 * 1024 //1MB
+)
+
+func TestTransition(t *testing.T) {
+	gob.Register(&entry.Entry{})
+
+	// build an old cache
+	echan := make(chan *entry.Entry, 8)
+	bchan := make(chan []*entry.Entry, 8)
+	ts := entry.Now()
+
+	tmp, err := ioutil.TempFile("", "bolttest")
+	if err != nil {
+		t.Fatal(err)
+	}
+	tmp.Close()
+	defer os.Remove(tmp.Name())
+
+	ic, err := NewIngestCache(IngestCacheConfig{
+		FileBackingLocation: tmp.Name(),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// write out some fake tags
+	if err := ic.UpdateStoredTagList([]string{"a", "b", "c"}); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := ic.Start(echan, bchan); err != nil {
+		t.Fatal(err)
+	}
+
+	for i := 0; i < 10; i++ {
+		ent := &entry.Entry{
+			TS:   ts,
+			SRC:  net.ParseIP("127.0.0.1"),
+			Tag:  entry.EntryTag(0),
+			Data: []byte("test data"),
+		}
+		echan <- ent
+	}
+
+	if err := ic.Stop(); err != nil {
+		t.Fatal(err)
+	}
+
+	if err := ic.Close(); err != nil {
+		t.Fatal(err)
+	}
+
+	// build a MuxerConfig and transition
+	err = boltTransition(MuxerConfig{
+		CachePath: tmp.Name(),
+	})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// tmp is now a directory bcause of the transition
+	defer os.RemoveAll(tmp.Name())
```

Comment:
why are there two defers?  one to Remove and one to RemoveAll ?

Reply:
tmp starts as a regular file, and the name becomes a directory under the transition, so we ignore what will be an erroneous os.Remove and add a RemoveAll for the new cache.